Files/Views/Pages/PropertiesDetails.xaml.cs

```diff
@@ -31,71 +31,41 @@ protected override void Properties_Loaded(object sender, RoutedEventArgs e)
             }
         }
 
-        private void SetOverviewVisibilities()
-        {
-            var name = ViewModel.ItemName.Split(".");
-            var extension = name[name.Length - 1].ToLower();
-
-            if (extension.Contains("png") || extension.Contains("jpg") || extension.Contains("gif") || extension.Contains("jpeg"))
-            {
-                OverviewImage.Visibility = Visibility.Visible;
-            }
-        }
-
-        private string GetStringArray(object array)
+        /// <summary>
+        /// Returns false if the operation was cancelled
+        /// </summary>
+        /// <param name="item"></param>
+        /// <returns></returns>
+        public async Task<bool> SaveChangesAsync()
         {
-            if (array == null || !(array is string[]))
+            try
             {
-                return "";
+                await (BaseProperties as FileProperties).SyncPropertyChangesAsync();
             }
-
-            var str = "";
-            foreach (var i in array as string[])
+            catch (Exception error)
             {
-                str += string.Format("{0}; ", i);
-            }
+                var dialog = new PropertySaveError();
 
-            return str;
-        }
-
-        private void SetStringArray(string val, string key)
-        {
-            ViewModel.SystemFileProperties_RW[key] = val.Split("; ");
-        }
+                switch (await dialog.ShowAsync())
+                {
+                    case ContentDialogResult.Primary:
+                        SaveChangesAsync();
```

Comment:
Recursive call?

Reply:
Yeah. That's if the user presses the retry button.
<img src="https://user-images.githubusercontent.com/59544401/101062978-22ef7280-3547-11eb-9fb2-d170bdfdf8e8.png" width="300"/>

Reply:
Should we await this call?

Reply:
@tsvietOK Is there a better way to do this?

Reply:
What about Infinite loop?

Reply:
Kind of
```
public async Task<bool> TrySaveChangesAsync()
{
    while (true)
    {
        try
        {
            await (BaseProperties as FileProperties).SyncPropertyChangesAsync();
        }
        catch
        {
            var dialog = new PropertySaveError();

            switch (await dialog.ShowAsync())
            {
                case ContentDialogResult.Primary:
                    break;
                case ContentDialogResult.Secondary:
                case ContentDialogResult.None:
                    return true;
                default:
                    break;
            }
        }
        break;
    }

    return true;
}

```